src/App.js

```diff
@@ -1,23 +1,49 @@
-import logo from './logo.svg';
 import './App.css';
+import foods from './foods.json';
+import { useState } from 'react';
+import FoodListComplete from './components/FoodListComplete';
+import AddFoodForm from './components/AddFoodForm';
+import Search from "./components/Search";
 
 function App() {
+  const [food, setFood] = useState(foods);
+  const [findFood, setFindFood] = useState(foods);
+  /*     const [showFood, setShowFood] = useState(foods); */
+
+  const searchFood = (search) => {
+    let filteredFood = food.filter((dish) =>
+      dish.name.toLowerCase().startsWith(search.toLowerCase())
+    );
+    setFindFood(filteredFood);
+  };
+
+  const addToList = (dish) => {
+    const foodCopy = [dish, ...food];
+    setFood(foodCopy);
+    setFindFood(foodCopy);
+  };
+
+
+  const deleteFood = (foodId) => {
+    const filteredFood = foods.filter((dish) => {
+      return foods._id !== foodId;
+    });
+    setFood(filteredFood);
```

Comment:
Your delete food is not working because:
- you're filtering through `foods` (json file) instead of `food` (variable);
- there is no `_id` in food object to compare, you can use `name` instead;
- you're only updating `food` with `setFood` you need to update `findFood` as well;
- you're passing the `deleteFood()` function as `props` to `FoodListComplete` but inside this component you're not using this function, you're using a copy of the function that you pasted inside the component;

App.js:
```
  const deleteFood = (foodId) => {
    const filteredFood = food.filter((dish) => {
      return dish.name !== foodId.name });
    setFood(filteredFood);
    setFindFood(filteredFood);
  };
```

FoodListComplete Component:
```
function FoodListComplete(props) {
  const { deleteFood, findFood } = props;

  return (
    <div>
       ( ... )
      <button onClick={() => deleteFood(dish)}>Delete</button>
    </div>
  );
}
export default FoodListComplete;
```